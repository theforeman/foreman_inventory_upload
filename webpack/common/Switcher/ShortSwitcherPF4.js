import React from 'react';
import PropTypes from 'prop-types';
import { Switch, Grid, GridItem } from '@patternfly/react-core';
import { HelpLabel } from './HelpLabel';
import './ShortSwitcherPF4.scss';

const ShortSwitcherPF4 = ({ id, label, tooltip, isChecked, onChange }) => (
  <Grid className="foreman-short-switcher">
    <GridItem span={9} className="switcher-label">
      {label}
      <HelpLabel text={tooltip} id={id} className="switcher-help-label" />
    </GridItem>
    <GridItem span={1} />
    <GridItem span={2}>
      <Switch
        id={`rh-cloud-switcher-${id}`}
        isChecked={isChecked}
        onChange={onChange}
      />
    </GridItem>
  </Grid>
);

ShortSwitcherPF4.propTypes = {
  id: PropTypes.string.isRequired,
  label: PropTypes.string,
  tooltip: PropTypes.string,
  isChecked: PropTypes.bool,
  onChange: PropTypes.func.isRequired,
};

ShortSwitcherPF4.defaultProps = {
  label: null,
  tooltip: null,
  isChecked: true,
};

export default ShortSwitcherPF4;
