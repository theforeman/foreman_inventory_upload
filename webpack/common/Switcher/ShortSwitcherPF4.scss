.rh-cloud-page {
  .foreman-short-switcher {
    margin-bottom: 5px;

    .switcher-help-label {
      padding-left: 5px;
    }

    .switcher-label {
      white-space: nowrap;
    }
  }
}
