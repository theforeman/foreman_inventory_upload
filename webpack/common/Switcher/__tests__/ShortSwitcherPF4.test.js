import { testComponentSnapshotsWithFixtures } from '@theforeman/test';
import ShortSwitcherPF4 from '../ShortSwitcherPF4';

const fixtures = {
  'should return insights cloud Url': {
    id: 'some-id',
    tooltip: 'some-text',
    label: 'some-label',
    onChange: jest.fn(),
  },
};

describe('InsightsCloudSync helpers', () =>
  testComponentSnapshotsWithFixtures(ShortSwitcherPF4, fixtures));
