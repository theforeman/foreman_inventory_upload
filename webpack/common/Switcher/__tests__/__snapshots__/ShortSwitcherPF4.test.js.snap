// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InsightsCloudSync helpers should return insights cloud Url 1`] = `
<Grid
  className="foreman-short-switcher"
>
  <GridItem
    className="switcher-label"
    span={9}
  >
    some-label
    <HelpLabel
      className="switcher-help-label"
      id="some-id"
      text="some-text"
    />
  </GridItem>
  <GridItem
    span={1}
  />
  <GridItem
    span={2}
  >
    <Switch
      aria-label=""
      id="rh-cloud-switcher-some-id"
      isChecked={true}
      isDisabled={false}
      onChange={[MockFunction]}
    />
  </GridItem>
</Grid>
`;
